File: docs/breez-sdk/snippets/flutter/lib/fiat_currencies.dart
import 'package:breez_sdk_spark_flutter/breez_sdk_spark.dart';

Future<ListFiatCurrenciesResponse> listFiatCurrencies(BreezClient client) async {
  // ANCHOR: list-fiat-currencies
  ListFiatCurrenciesResponse response = await client.listFiatCurrencies();
  // ANCHOR_END: list-fiat-currencies
  return response;
}

Future<ListFiatRatesResponse> listFiatRates(BreezClient client) async {
  // ANCHOR: list-fiat-rates
  ListFiatRatesResponse response = await client.listFiatRates();
  // ANCHOR_END: list-fiat-rates
  return response;
}

File: docs/breez-sdk/snippets/swift/BreezSdkSnippets/Sources/FiatCurrencies.swift
import BreezSdkSpark
import Foundation

func listFiatCurrencies(client: BreezClient) async throws -> ListFiatCurrenciesResponse {
    // ANCHOR: list-fiat-currencies
    let response = try await client.fiat().currencies()
    // ANCHOR_END: list-fiat-currencies
    return response
}

func listFiatRates(client: BreezClient) async throws -> ListFiatRatesResponse {
    // ANCHOR: list-fiat-rates
    let response = try await client.fiat().rates()
    // ANCHOR_END: list-fiat-rates
    return response
}
